src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -373,27 +375,40 @@ private void UpdateReferences(TransactionOperationContext indexContext, string c
             if (CurrentIndexingScope.Current.ReferencesByCollection != null &&
                 CurrentIndexingScope.Current.ReferencesByCollection.TryGetValue(collection, out var values))
             {
-                _indexStorage.ReferencesForDocuments.WriteReferencesForSingleCollection(collection, values, indexContext.Transaction);
+                using (stats.For(IndexingOperation.Storage.UpdateReferences))
+                    _indexStorage.ReferencesForDocuments.WriteReferencesForSingleCollection(collection, values, indexContext.Transaction);
                 values.Clear();
             }
 
             if (CurrentIndexingScope.Current.ReferencesByCollectionForCompareExchange != null &&
                 CurrentIndexingScope.Current.ReferencesByCollectionForCompareExchange.TryGetValue(collection, out values))
             {
-                _indexStorage.ReferencesForCompareExchange.WriteReferencesForSingleCollection(collection, values, indexContext.Transaction);
+                using (stats.For(IndexingOperation.Storage.UpdateReferences))
+                    _indexStorage.ReferencesForCompareExchange.WriteReferencesForSingleCollection(collection, values, indexContext.Transaction);
                 values.Clear();
             }
         }
 
         private IEnumerable<IndexItem> GetItemsFromCollectionThatReference(QueryOperationContext queryContext, TransactionOperationContext indexContext,
-            string collection, Reference referencedItem, long lastIndexedEtag, HashSet<string> indexed, ReferencesState.ReferenceState referenceState)
+            string collection, Reference referencedItem, long lastIndexedEtag, HashSet<string> indexed, ReferencesState.ReferenceState referenceState, IndexingStatsScope stats)
         {
             var lastProcessedItemId = referenceState?.GetLastProcessedItemId(referencedItem);
+            List<Slice> keysToRemove = null;
```

Comment:
Can we indicated somehow it's for "legacy handling" purposes?

Reply:
Removed this in favor of a different change